api/src/main/java/org/openmrs/module/fhir/api/strategies/relatedperson/RelatedPersonStrategy.java

```diff
@@ -27,6 +31,33 @@ public RelatedPerson getRelatedPerson(String uuid) {
 		return FHIRRelatedPersonUtil.generateRelationshipObject(omrsRelationship);
 	}
 
+	/**
+	 * @see org.openmrs.module.fhir.api.strategies.relatedperson.GenericRelatedPersonStrategy#searchRelatedPersonByIdentifier(String)
+	 */
+	@Override
+	public List<RelatedPerson> searchRelatedPersonByIdentifier(String identifierValue) {
+		List<RelatedPerson> relatedPersonList = new ArrayList<>();
+		List<Relationship> relationships = new ArrayList<>();
+		PatientService patientService = Context.getPatientService();
+		List<org.openmrs.Patient> patientList = patientService.getPatients(null,
```

Comment:
Could you be knowing why we had no tests that failed even when we were passing an identifier where we were supposed to pass a name?

Reply:
Yeah
This is why 

`if (name == null) {
                name = identifier;
            } else {
                identifier = name;
            }`

It's implemented in openmrs core api

Reply:
Nice finding! 👍